### PR TITLE
feat: add support for managing multiple databases

### PR DESCRIPTION
**Pull request by NoyException**

### CODE REVIEW

**Review by TianyuZhang1214 — CHANGES_REQUESTED** [review]
Please add tests for this PR.

File: catalog/provider.go [hunk 1]
```diff
@@ -93,35 +140,126 @@ func NewDBProvider(dataDir, dbFile string) (*DatabaseProvider, error) {
...
-		catalogName:               name,
-		dataDir:                   dataDir,
-		dbFile:                    dbFile,
-		dsn:                       dsn,
-		externalProcedureRegistry: sql.NewExternalStoredProcedureRegistry(), // This has no effect, just to satisfy the upper layer interface
-	}, nil
+	return nil
+}
+
+func (prov *DatabaseProvider) CreateCatalog(dbFile string) (bool, error) {
```
Comment (fanyang01) [inline catalog/provider.go:165]: Maybe we should simplify the `*Catalog` APIs to accept a catalog name without the `.db` suffix.

File: catalog/internal_tables.go [hunk 2]
```diff
@@ -12,6 +12,9 @@ type InternalTable struct {
 }
 
 func (it *InternalTable) QualifiedName() string {
+	//if it.Schema == "__sys__" {
```
Comment (TianyuZhang1214) [inline catalog/internal_tables.go:15]: remove unnecessary comment here

File: catalog/provider.go [hunk 3]
```diff
@@ -37,31 +42,37 @@ var _ configuration.DataDirProvider = (*DatabaseProvider)(nil)
...
 }
 
-func NewDBProvider(dataDir, dbFile string) (*DatabaseProvider, error) {
-	dbFile = strings.TrimSpace(dbFile)
-	name := ""
-	dsn := ""
-	if dbFile == "" {
-		// in-memory mode, mainly for testing
-		name = "memory"
+func NewDBProvider(defaultTimeZone, dataDir, defaultDB string) (prov *DatabaseProvider, err error) {
```
Comment (TianyuZhang1214) [inline catalog/provider.go:52]: It seems we don't have strategy to scan all .db files under the data dir and attach them to current datbase?

File: catalog/provider.go [hunk 4]
```diff
@@ -20,13 +22,16 @@ import (
 
 type DatabaseProvider struct {
 	mu                        *sync.RWMutex
+	defaultTimeZone           string
 	connector                 *duckdb.Connector
 	storage                   *stdsql.DB
-	catalogName               string
+	pool                      *ConnectionPool // TODO(Noy): Merge into the provider
```
Comment (TianyuZhang1214) [inline catalog/provider.go:28]: It's better to make it an issue of refactoring.

**Review by TianyuZhang1214 — CHANGES_REQUESTED** [review]
Great job!

**Review by TianyuZhang1214 — APPROVED** [review]
LGTM

**Pull request merged**